### DISCUSSION TITLE
I am a student who learn cvxpy in few days,I meet a problem which I can't solve.

### DISCUSSION THREAD

**Discussion by linfoxcion** [cvxpy/cvxpy#1907] [Q&A]
### I want solve this problem:
`  t=cp.Variable()
    constraints=[t>=-1,t<=10]
    prob2 = cp.Problem(cp.Minimize(cp.power(cp.inv_pos(t)-20)), constraints)
    prob2.solve(solver='MOSEK')
    print("result:", prob2.value)
    return prob2.value`
however,terminal show that: `  raise DCPError(
cvxpy.error.DCPError: Problem does not follow DCP rules. Specifically:
The objective is not DCP. Its following subexpressions are not:
power(power(var2, -1.0) + -20.0, 2.0)`
Can I have your answer? Thank you

**Comment by linfoxcion** [comment]
correct::: prob2 should be `_prob2 = cp.Problem(cp.Minimize(cp.power(cp.inv_pos(t)-20),2), constraints)_`

**Reply by linfoxcion** [reply]
I don't know why not the expression follow DCP,the subfunction 1/t is convex,1/t-20 also convex, x^2 is convex

**Reply by phschiele** [reply]
The problem is the `-20`. According to the DCP rules, we can compose convex functions with convex functions for arguments in which they are increasing, and with concave functions for arguments in which they are decreasing (affine works for both).
x^2 is convex, as you point out, and increasing only for x>=0. We know that `cp.inv_pos(t)` is convex and positive, so the composition works according to the DCP rules. As soon as we subtract the 20, we no longer know the sign, so the DCP rules are violated.
You should check out the [short course](https://stanford.edu/~boyd/papers/cvx_short_course.html) for a more detailed introduction.

**Reply by linfoxcion** [reply]
Thank you so much,professor.If I want solve the problem:`(1/t-20)^2, constrain=[t>=1,t<=10],`,how to write the code?
